client/components/drop-zone/index.jsx

```diff
@@ -144,6 +153,25 @@ export const DropZone = React.createClass( {
 			// from tracked nodes since another "real" event will be triggered.
 			this.dragEnterNodes = without( this.dragEnterNodes, window );
 		}
+
+		this.toggleDropZoneReduxState( this.state.isDraggingOverDocument || this.state.isDraggingOverElement );
+	},
+
+	toggleDropZoneReduxState( isVisible ) {
+		/**
+		 * The DropZone component is included in components/libraries which are not yet reduxified,
+		 * for which `connect` ( from `react-redux` ) complains that it can't find `store`
+		 * in the current context.
+		 *
+		 * For places where the reduxification has happened, this will work.
+		 */
+		if ( this.context.store ) {
+			if ( isVisible ) {
+				this.context.store.dispatch( showDropZone() );
+			} else {
+				this.context.store.dispatch( hideDropZone() );
+			}
+		}
```

Comment:
Let's avoid solutions like this. The mising store could (and should) be solved by the introduction of `<ReduxProvider />` in the relevant places.

Keep in mind, there was a time when there were almost no `<ReduxProvider />`s around. Thanks to [some folks](https://github.com/Automattic/wp-calypso/pull/7198), this is no longer the case, so there should be fewer places  where they're still missing -- and thus, less work needed to introduce them there.